cfgov/unprocessed/js/molecules/Notification.js

```diff
@@ -0,0 +1,148 @@
+  var SUCCESS = 'success';
+  var WARNING = 'warning';
+  var ERROR = 'error';
+
+  var _dom =
+    atomicCheckers.validateDomElement( element, BASE_CLASS, 'Notification' );
+  var _icon = _dom.querySelector( '.' + BASE_CLASS + '_icon' );
+  var _content = _dom.querySelector( '.' + BASE_CLASS + '_content' );
+
+  var _currentType;
+
+  /**
+   * @returns {Object} The Notification instance.
+   */
+  function init() {
+    // Check and set default type of notification.
+    var classList = _dom.classList;
+    if ( classList.contains( BASE_CLASS + '__' + SUCCESS ) ) {
+      _currentType = SUCCESS;
+    } else if ( classList.contains( BASE_CLASS + '__' + WARNING ) ) {
+      _currentType = WARNING;
+    } else if ( classList.contains( BASE_CLASS + '__' + ERROR ) ) {
+      _currentType = ERROR;
+    }
+
+    return this;
+  }
+
+  /**
+   * @param {number} type The notifiation type.
+   * @param {string} messageText The content of the notifiation message.
+   * @param {string|HTMLNode} explanationText
+   *   The content of the notifiation explanation.
+   * @returns {Object} The Notification instance.
+   */
+  function setContent( type, messageText, explanationText ) {
+    _setType( type );
+    var content = '<p class="h4">' +
+                  messageText +
+                  '</p>';
+    if ( typeof explanationText !== 'undefined' ) {
+      content += '<p class="h4 m-notification_explanation">' +
+                 explanationText +
+                 '</p>';
+    }
+    _content.innerHTML = content;
+
+    return this;
+  }
+
+  /**
+   * @param {number} type The notifiation type.
+   * @returns {Object} The Notification instance.
+   */
+  function _setType( type ) {
+    // If type hasn't changed, return.
+    if ( _currentType === type ) {
+      return this;
+    }
+
+    var classList = _dom.classList;
+    classList.remove( BASE_CLASS + '__' + SUCCESS );
+    classList.remove( BASE_CLASS + '__' + WARNING );
+    classList.remove( BASE_CLASS + '__' + ERROR );
+
+    _icon.className = BASE_CLASS + '_icon ' + 'cf-icon';
+    var iconClassList = _icon.classList;
+
+    if ( type === SUCCESS ) {
+      classList.add( BASE_CLASS + '__' + SUCCESS )
+      iconClassList.add( BASE_CLASS + '_icon__' + SUCCESS );
+      iconClassList.add( 'cf-icon-approved-round' );
+    } else if ( type === WARNING ) {
+      classList.add( BASE_CLASS + '__' + WARNING )
+      iconClassList.add( BASE_CLASS + '_icon__' + WARNING );
+      iconClassList.add( 'cf-icon-error-round' );
+    } else if ( type === ERROR ) {
+      classList.add( BASE_CLASS + '__' + ERROR )
+      iconClassList.add( BASE_CLASS + '_icon__' + ERROR );
+      iconClassList.add( 'cf-icon-delete-round' );
+    } else {
+      throw new Error( type + ' is not a supported notification type!' );
+    }
+
+    _currentType = type;
+
+    return this;
+  }
+
+  /**
+   * @returns {Object} The Notification instance.
+   */
+  function show() {
+    if ( _currentType === ERROR ||
```

Comment:
No, I felt like it was slightly more readable stacked, but one line is fine too since it messes up the nesting of the conditional a bit. Fixed in https://github.com/cfpb/cfgov-refresh/commit/4fb28428fa39389698ba68f49129d47a450f2c53